configure.py

```diff
@@ -633,9 +647,11 @@ def has_re2c():
     print('bootstrap complete.  rebuilding...')
     if platform.is_windows():
         bootstrap_exe = 'ninja.bootstrap.exe'
-        if os.path.exists(bootstrap_exe):
-            os.unlink(bootstrap_exe)
-        os.rename('ninja.exe', bootstrap_exe)
-        subprocess.check_call('ninja.bootstrap.exe%s' % verbose, shell=True)
+        final_exe = 'ninja.exe'
     else:
-        subprocess.check_call('./ninja%s' % verbose, shell=True)
+        bootstrap_exe = './ninja_bootstrap'
+        final_exe = './ninja'
+    if os.path.exists(bootstrap_exe):
+        os.unlink(bootstrap_exe)
+    os.rename(final_exe, bootstrap_exe)
+    subprocess.check_call(bootstrap_exe + verbose, shell=True)
```

Comment:
Maybe it would help me understand better if you described your workflow as a ninja developer -- maybe it's different from how I've been working. I have a script that runs `./configure.py --bootstrap`, followed by `./ninja ninja_test` and `./ninja_test`. To me, the impact of not overwriting the bootstrap first-pass result is just another file on disk. With the rename of the first executable as `ninja_bootstrap` and the self-built result as `ninja`, no one will run the first one by accident.

It's looking like the overwrite is definitely a problem on AIX, and my teammates suggest that isn't a great idea other platforms, so I'd like to understand what your usage pattern is.